featuretools/tests/primitive_tests/test_max_consecutive.py

```diff
@@ -11,230 +12,211 @@
+        array = pd.Series([1.0, -1.0, 2.0, 4.0, 5.0], dtype=dtype)
         assert primitive_func(array) == 3
 
-    def test_all_true(self):
-        primitive_instance = self.primitive()
-        primitive_func = primitive_instance.get_function()
-        array = pd.Series([True, True, True, True])
-        assert primitive_func(array) == 4
-
-    def test_all_false(self):
-        primitive_instance = self.primitive()
+    def test_with_nan(self, dtype):
+        if dtype == "int64":
+            pytest.skip("nans not supported in int64")
+        primitive_instance = MaxConsecutivePositives()
         primitive_func = primitive_instance.get_function()
-        array = pd.Series([False, False, False])
-        assert primitive_func(array) == 0
-
-    def test_with_nan(self):
-        primitive_instance = self.primitive()
-        primitive_func = primitive_instance.get_function()
-        array = pd.Series([True, np.nan, True, False])
+        array = pd.Series([1, np.nan, 2, -3], dtype=dtype)
         assert primitive_func(array) == 2
 
-    def test_with_nan_skipna(self):
-        primitive_instance = self.primitive(skipna=False)
+    def test_with_nan_skipna(self, dtype):
+        if dtype == "int64":
+            pytest.skip("nans not supported in int64")
+        primitive_instance = MaxConsecutivePositives(skipna=False)
         primitive_func = primitive_instance.get_function()
-        array = pd.Series([True, np.nan, True, False])
+        array = pd.Series([1, np.nan, 2, -3], dtype=dtype)
         assert primitive_func(array) == 1
 
-    def test_all_nan(self):
-        primitive_instance = self.primitive()
+    def test_all_nan(self, dtype):
+        if dtype == "int64":
+            pytest.skip("nans not supported in int64")
+        primitive_instance = MaxConsecutivePositives()
         primitive_func = primitive_instance.get_function()
-        array = pd.Series([np.nan, np.nan, np.nan, np.nan])
+        array = pd.Series([np.nan, np.nan, np.nan, np.nan], dtype=dtype)
         assert np.isnan(primitive_func(array))
 
-    def test_all_nan_skipna(self):
-        primitive_instance = self.primitive(skipna=True)
+    def test_all_nan_skipna(self, dtype):
+        if dtype == "int64":
+            pytest.skip("nans not supported in int64")
+        primitive_instance = MaxConsecutivePositives(skipna=True)
         primitive_func = primitive_instance.get_function()
-        array = pd.Series([np.nan, np.nan, np.nan, np.nan])
+        array = pd.Series([np.nan, np.nan, np.nan, np.nan], dtype=dtype)
         assert np.isnan(primitive_func(array))
 
 
+@pytest.mark.parametrize("dtype", ["float64", "int64"])
 class TestMaxConsecutiveZeros:
-    primitive = MaxConsecutiveZeros
-
-    def test_regular(self):
-        primitive_instance = self.primitive()
+    def test_regular(self, dtype):
+        if dtype == "int64":
+            pytest.skip("floats not supported in int64")
```

Comment:
floats -> nans

Reply:
No this was intentional message. I made a better one
https://github.com/alteryx/featuretools/pull/2420/commits/a014517e40ce6dde0475b6c242bfe4566b183556